### DISCUSSION TITLE
Help with implementing - Failed on 6.2.5

### DISCUSSION THREAD

**Discussion by CryingEagle1** [alivx/CIS-Ubuntu-20.04-Ansible#49] [Q&A]
Dear alivx/community,

Currently I am learning and trying to implement the CIS standard on a machine but it keeps getting stuck at 6.2.5 [fix] (6.2.5 Ensure users home directories permissions are 750 or more restrictive - fix]) whilst trying to run it. I haven't been able to make it work and I was wondering if you had any solution/tips for me on this error. (maybe I even overlooked/missed something?)

The specific error that occurs:
> failed: [client-ip-address] (item=) => {"ansible_loop_var": "item", "changed": false, "item": "", "msg": "file () is absent, cannot continue", "path": "", "state": "absent"}

I hope I provided you with enough information for you to be able to point me in the right direction. :)
Many Thanks,
CryingEagle1

**Comment by alivx** [comment]
Can you run this script and provide to me the output,
`grep -E -v '^(halt|sync|shutdown)' /etc/passwd | awk -F: '($7 != "'"$(which nologin)"'" && $7 != "/bin/false") { print $1 " " $6 }' | while read user dir; do
    if [ ! -d "$dir" ]; then
        echo "The home directory ($dir) of user $user does not exist."
    else
        dirperm=$(ls -ld $dir | cut -f1 -d" ")
        if [ $(echo $dirperm | cut -c6) != "-" ]; then
            echo "$dir" >/tmp/.6.2.5
        fi
        if [ $(echo $dirperm | cut -c8) != "-" ]; then
            echo "$dir" >>/tmp/.6.2.5
        fi
        if [ $(echo $dirperm | cut -c9) != "-" ]; then
            echo "$dir" >>/tmp/.6.2.5
        fi
        if [ $(echo $dirperm | cut -c10) != "-" ]; then
            echo "$dir" >>/tmp/.6.2.5
        fi
    fi
    cat /tmp/.6.2.5 | sort -n | uniq
done`

Plus, can you add `-vvvv` to ansible run command in order to see the debug output.

**Reply by CryingEagle1** [reply]
Executing the script gave me an error: syntax error near unexpected token `then'.
I kind of broke down the script in smaller parts and I think you are looking for the result of ls -ld on all regular users.
drwx------ 5 root root 4096 May 17 07:43 /root/
and
drwxr-x--- 5 ubuntu ubuntu 4096 May 17 08:19 /home/ubuntu
Would be the result of that.

**Reply by alivx** [reply]
It's seems the text format gone.

Please run the file
https://raw.githubusercontent.com/alivx/CIS-Ubuntu-20.04-Ansible/master/files/6_2_5.sh

-

**Reply by CryingEagle1** [reply]
I tried to run the script on multiple vm's now. On my main testing machine the .6.2.5 file only has read perms -rw-r----- 1 root root 0 May 25 12:05 /tmp/.6.2.5 + every time I tried to run the script after I deleted the file it still got the same permissions.
On the non patched machine running the script caused it to output the user folder like it is intended. The permissions of .6.2.5 on that machine are -rw-r--r-- 1 root root 26 May 25 11:49 /tmp/.6.2.5. I have tried to change the permissions of /tmp on the main machine but that didn't work for me.
